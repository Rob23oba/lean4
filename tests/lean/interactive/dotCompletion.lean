namespace Outer

def hi : Nat := 42

namespace Inner

end Inner

end Outer

#print Outer.
           --^ textDocument/completion

syntax Outer. := "hi"
           --^ textDocument/completion

syntax "test_me " Outer.stx : command


-- TODO: don't suggest `hi`
namespace Outer.
              --^ textDocument/completion
