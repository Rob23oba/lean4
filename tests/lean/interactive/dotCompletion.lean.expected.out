{"textDocument": {"uri": "file:///dotCompletion.lean"},
 "position": {"line": 10, "character": 13}}
{"items":
 [{"label": "Inner",
   "kind": 9,
   "detail": "namespace",
   "data":
   {"params":
    {"textDocument": {"uri": "file:///dotCompletion.lean"},
     "position": {"line": 10, "character": 13}},
    "cPos": 0}},
  {"label": "hi",
   "kind": 21,
   "data":
   {"params":
    {"textDocument": {"uri": "file:///dotCompletion.lean"},
     "position": {"line": 10, "character": 13}},
    "id": {"const": {"declName": "Outer.hi"}},
    "cPos": 0}}],
 "isIncomplete": true}
{"textDocument": {"uri": "file:///dotCompletion.lean"},
 "position": {"line": 13, "character": 13}}
{"items":
 [{"label": "Inner",
   "kind": 9,
   "detail": "namespace",
   "data":
   {"params":
    {"textDocument": {"uri": "file:///dotCompletion.lean"},
     "position": {"line": 13, "character": 13}},
    "cPos": 0}},
  {"label": "hi",
   "kind": 21,
   "data":
   {"params":
    {"textDocument": {"uri": "file:///dotCompletion.lean"},
     "position": {"line": 13, "character": 13}},
    "id": {"const": {"declName": "Outer.hi"}},
    "cPos": 0}}],
 "isIncomplete": true}
{"textDocument": {"uri": "file:///dotCompletion.lean"},
 "position": {"line": 17, "character": 16}}
{"items":
 [{"label": "Inner",
   "kind": 9,
   "detail": "namespace",
   "data":
   {"params":
    {"textDocument": {"uri": "file:///dotCompletion.lean"},
     "position": {"line": 17, "character": 16}},
    "cPos": 0}},
  {"label": "hi",
   "kind": 21,
   "data":
   {"params":
    {"textDocument": {"uri": "file:///dotCompletion.lean"},
     "position": {"line": 17, "character": 16}},
    "id": {"const": {"declName": "Outer.hi"}},
    "cPos": 0}}],
 "isIncomplete": true}
